x-pack/plugins/inference/server/chat_complete/adapters/bedrock/bedrock_claude_adapter.ts

```diff
@@ -0,0 +1,138 @@
+import { Message, MessageRole } from '../../../../common/chat_complete';
+import { createInferenceInternalError } from '../../../../common/errors';
+import { ToolChoiceType, type ToolOptions } from '../../../../common/chat_complete/tools';
+import { InferenceConnectorAdapter } from '../../types';
+import type { BedRockMessage, BedrockToolChoice } from './types';
+import {
+  BedrockChunkMember,
+  serdeEventstreamIntoObservable,
+} from './serde_eventstream_into_observable';
+import { processCompletionChunks } from './process_completion_chunks';
+
+export const bedrockClaudeAdapter: InferenceConnectorAdapter = {
+  chatComplete: ({ executor, system, messages, toolChoice, tools }) => {
+    const connectorInvokeRequest: InvokeAIActionParams = {
+      system,
+      messages: messagesToBedrock(messages),
+      tools: toolsToBedrock(tools),
+      toolChoice: toolChoiceToBedrock(toolChoice),
+      temperature: 0,
+      stopSequences: ['\n\nHuman:'],
+    };
+
+    return from(
+      executor.invoke({
+        subAction: 'invokeStream',
+        subActionParams: connectorInvokeRequest,
+      })
+    ).pipe(
+      switchMap((response) => {
+        const readable = response.data as Readable;
+        return serdeEventstreamIntoObservable(readable);
+      }),
+      tap((eventData) => {
+        if ('modelStreamErrorException' in eventData) {
+          throw createInferenceInternalError(eventData.modelStreamErrorException.originalMessage);
+        }
+      }),
+      filter((value): value is BedrockChunkMember => {
+        return 'chunk' in value && value.chunk?.headers?.[':event-type']?.value === 'chunk';
+      }),
+      map((message) => {
+        return parseSerdeChunkMessage(message.chunk);
+      }),
+      processCompletionChunks()
+    );
+  },
+};
+
+const toolChoiceToBedrock = (
+  toolChoice: ToolOptions['toolChoice']
+): BedrockToolChoice | undefined => {
+  if (toolChoice === ToolChoiceType.required) {
+    return {
+      type: 'any',
+    };
+  } else if (toolChoice === ToolChoiceType.auto) {
+    return {
+      type: 'auto',
+    };
+  } else if (typeof toolChoice === 'object') {
+    return {
+      type: 'tool',
+      name: toolChoice.function,
+    };
+  }
+  // ToolChoiceType.none is not supported by claude.
+  return undefined;
+};
+
+const toolsToBedrock = (tools: ToolOptions['tools']) => {
+  return tools
+    ? Object.entries(tools).map(([toolName, toolDef]) => {
+        return {
+          name: toolName,
+          description: toolDef.description,
+          input_schema: toolDef.schema ?? {
+            type: 'object' as const,
+            properties: {},
+          },
+        };
+      })
+    : undefined;
+};
+
+const messagesToBedrock = (messages: Message[]): BedRockMessage[] => {
+  return messages.map<BedRockMessage>((message) => {
+    switch (message.role) {
+      case MessageRole.User:
+        return {
+          role: 'user' as const,
+          // content: message.content,
```

Comment:
why commented?

Reply:
I initially used the `content` field for user message, then I decided to use `rawContent` (which gets mapped directly to `content` in the connector without transformations) for all type of messages.

I just forgot to delete the commented bits, thanks